Revised conditional logical operators articles

pkulikov opened the pull request:
The code examples are much shortened and moved to the samples repo: dotnet/samples#442

pkulikov committed:
Revised conditional logical operators articles

pkulikov committed:
Tweaked the link text

BillWagner commented on docs/csharp/language-reference/operators/and-operator.md:
```diff
@@ -27,7 +27,7 @@ For integer types, the `&` operator computes the logical bitwise AND of its oper
 
 > [!NOTE]
-> The preceding example uses the binary literals that are [introduced](../../whats-new/csharp-7.md#numeric-literal-syntax-improvements) in C# 7.0 and [enhanced](../../whats-new/csharp-7-2.md#leading-underscores-in-numeric-literals) in C# 7.2.
+> The preceding example uses the binary literals that are [introduced in C# 7.0](../../whats-new/csharp-7.md#numeric-literal-syntax-improvements) and [enhanced in C# 7.2](../../whats-new/csharp-7-2.md#leading-underscores-in-numeric-literals).
```
I would remove "that are"

BillWagner reviewed (changes requested):
Thanks for putting this together @pkulikov 

I had one small text suggestion, because C# 7.0 has been around for a while.  :smile:

Once you make that change, I'll :shipit:

pkulikov committed:
Merge branch 'master' into revise-conditional-logical-operators

pkulikov committed:
Addressed feedback

pkulikov commented:
@BillWagner thanks, I've addressed your feedback. Also updated the same note in the article about the `~` operator, in which tweaked the first sentence a bit, as well.
